src/blocksparsearray/blockdiagonalarray.jl

```diff
@@ -12,9 +12,7 @@ const BlockSparseDiagonal{T,A<:AbstractBlockSparseVector{T}} = Diagonal{T,A}
 end
 
 function BlockDiagonal(blocks::AbstractVector{<:AbstractMatrix})
-  return BlockSparseArray(
-    Diagonal(blocks), (blockedrange(size.(blocks, 1)), blockedrange(size.(blocks, 2)))
-  )
+  return sparsemortar(Diagonal(blocks), size.(blocks, 1), size.(blocks, 2))
```

Comment:
I think this one is also secretly a constructor (`BlockSparseMatrix{T,A,Diagonal{A,V},Ax} where {T,A,V<:AbstractVector{A},Ax}`), so maybe this should also use the inner constructor?